src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfig.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types.Enums;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// A container that allows for compiler customization.
+    /// </summary>
+    public sealed class PowerFxConfig
```

Comment:
>PowerFxConfig [](http://example.com/codeflow?start=24&length=13)

Going forward - would we create one of these around an MsApp? 
We get lots of requests for tools doing intellisense over formulas in an msapp, but they can't create the context today.

Reply:
Would Document (eventually) use this? Or other backends?
Should we move to Core?

`AddFunction(ReflectionFunction function)` may be an issue (that's interpreter specific), but that could easily be an extension method in interpreter that calls  ` AddFunction(TexlFunction function)`

Reply:
Not sure on the future with document specifically, but other back ends would most likely want this, I've moved it Core with some extension methods.